src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -35,27 +35,32 @@
 
 (schema/defn clojurize :- Event
   [event :- WatchEvent
-   watch-path :- Path]
-  {:type (get event-type-mappings (.kind event))
-   :path (when-not (= StandardWatchEventKinds/OVERFLOW (.kind event))
-           (-> watch-path
-               (.resolve (.context event))
-               fs/file))})
-
-;; This is quite similar to the function above but it is more a direct
-;; conversion of the exact data available on a specific WatchEvent instance,
-;; only used for debugging.
-(defn clojurize-for-logging
-  [e]
-  {:context (.context e)
-   :count (.count e)
-   :kind (.kind e)})
-
-(defn pprint-events
-  [events]
-  (->> events
-       (map #(update % :path str))
-       ks/pprint-to-string))
+   watched-path :- Path]
+  (let [kind (get event-type-mappings (.kind event))
+        changed-path (when-not (= :unknown kind)
+                       (.context event))
+        full-path (when-not (= :unknown kind)
+                    (-> watched-path (.resolve changed-path) fs/file))]
+    {:type kind
+     :count (.count event)
+     :watched-path (fs/file watched-path)
+     :changed-path (fs/file changed-path)
+     :full-path full-path}))
+
+(schema/defn format-for-debugging
+  [{:keys [changed-path count type]} :- Event]
+  {:context changed-path
```

Comment:
I found `changed-path` more intuitive than `context` which is what it was prior to this PR.

`context` is the accessor's name on the Java object, this format was originally meant (if I'm reading the tea leaves correctly) to basically be a serialization of the WatchEvent interface for those familiar with it.

I've tried to consolidate the structures we pass around internally while retaining the same output to our consumers.

I changed `context` to `changed-path` because using the closest-names-to-java-members/what-was-there-before would have yielded something like: `watchable`, `context`, and `path`, which I thought were more intuitively referred to as `watched-path`, `changed-path`, and `full-path` (I mainly thought that `path` should be disambiguated).

I'm happy to either refer to them with the former listed keys, or to keep the latter keys and not translate them back. Do you have a preference?